polkadot/runtime/rococo/src/lib.rs

```diff
@@ -1307,6 +1307,52 @@ impl pallet_asset_rate::Config for Runtime {
 	type BenchmarkHelper = runtime_common::impls::benchmarks::AssetRateArguments;
 }
 
+#[frame_support::pallet]
+pub mod im_online_remover {
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config {}
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_initialize(n: BlockNumberFor<T>) -> Weight {
+			if RemoveAtBlock::<T>::get() == None {
+				RemoveAtBlock::<T>::set(Some(n));
+			}
+			Weight::zero()
+		}
+
+		fn offchain_worker(n: BlockNumberFor<T>) {
+			const DB_PREFIX: &[u8] = b"parity/im-online-heartbeat/";
+			if let Some(remove_at) = RemoveAtBlock::<T>::get() {
```

Comment:
I would move the actual deletion logic into the `im-online` pallet `migration.rs` into a function. 

I'm also not sure we need this `RemoveAtBlock` guarding. We could probably just call this extra function in the `offchain_worker` runtime api function below. We could check that if the passed header has the [`RuntimeEnvironmentUpgraded`](https://github.com/paritytech/polkadot-sdk/blob/9db92115d023689581e8cf49edb9500b0dcced53/substrate/primitives/runtime/src/generic/digest.rs#L108) digest for guarding the call to the function.